thunderfish/sortfishtype.py

```diff
@@ -0,0 +1,189 @@
+
+    ax[0].add_patch(Rectangle((t[start_in], min_amp),  # (x,y)
+                              t[end_in]-t[start_in],  # width
+                              max_amp + abs(min_amp),  # height
+                              edgecolor='black', facecolor='white', lw=3))
+
+    # Cosmetics
+    ax[0].set_title('Best window with peak detection', fontsize=fs+2)
+    ax[0].set_xlabel('Time [sec]', fontsize=fs)
+    ax[0].set_ylabel('Amplitude [a.u.]', fontsize=fs)
+    ax[0].tick_params(axis='both', which='major', labelsize=fs - 2)
+
+    # Second Plot: Best-Window inset
+    ax[1].set_xlim(t[start_in], t[end_in])  # just set proper xlim!
+
+    # Cosmetics
+    ax[1].set_title('Best window Inset. Fish-type suggestion is %s' % type_suggestion, fontsize=fs + 2)
+    ax[1].set_xlabel('Time [sec]', fontsize=fs)
+    ax[1].set_ylabel('Amplitude [a.u.]', fontsize=fs)
+    ax[1].tick_params(axis='both', which='major', labelsize=fs - 2)
+
+    # Third Plot: r-values and Fish-type threshold
+    r_vals = r_vals[r_vals > 0]
+    hist, bins = np.histogram(r_vals, bins=50)
+    width = 0.7 * (bins[1] - bins[0])
+    center = (bins[:-1] + bins[1:]) / 2
+    ax[2].bar(center, hist, align='center', facecolor='darkgreen', edgecolor='black', lw=2.5, alpha=0.7, width=width)
+    ax[2].plot([r_value_th, r_value_th], [0, np.max(hist)], '--k', lw=2.5, alpha=0.7)
+
+    # Cosmetics
+    ax[2].set_xlabel('Peak2Trough / Peak2Peak', fontsize=fs)
+    ax[2].set_ylabel('Counts', fontsize=fs)
+    ax[2].tick_params(axis='both', which='major', labelsize=fs - 2)
+    ax[2].set_title('Distribution of r-values', fontsize=fs + 2)
+
+    pass
+
+
+if __name__ == "__main__":
+    print("\nChecking sortfishtype module ...\n")
+    import sys
+    import matplotlib.pyplot as plt
+
+    if len(sys.argv) < 2:
+        # generate data:
+        print("Generating artificial waveform...")
+        rate = 40000.0
+        time = np.arange(0.0, 10.0, 1./rate)
+        f1 = 100.0
+        data0 = (0.5*np.sin(2.0*np.pi*f1*time)+0.5)**20.0
+        amf1 = 0.3
+        data1 = data0*(1.0-np.cos(2.0*np.pi*amf1*time))
+        data1 += 0.2
+        f2 = f1*2.0*np.pi
+        data2 = 0.1*np.sin(2.0*np.pi*f2*time)
+        amf3 = 0.15
+        data3 = data2*(1.0-np.cos(2.0*np.pi*amf3*time))
+        #data = data1+data3
+        data = data2
+        data += 0.01*np.random.randn(len(data))
+
+    else:  # load data given by the user
+        import dataloader as dl
+
+        file_path = sys.argv[1]
+        print("loading %s ...\n" % file_path)
+        data, rate, unit = dl.load_data(sys.argv[1], 0)
+        time = np.arange(len(data)) / rate
+
+    # run fish-type detector
+    type_suggestion, r_val = type_detector(data, rate, plot_data_func=plot_type_detection)
+    if len(sys.argv) >= 2:
+        filename = file_path.split('/')[-1]
+        title = 'Fish # %s is %s' % (filename, type_suggestion)
+        fig = plt.gcf()
+        fig.canvas.set_window_title(title)
+    plt.tight_layout()
```

Comment:
the plotting should go into a separate module as pointed out above for the plot function.